rcore-fs/src/vfs.rs

```diff
@@ -95,6 +95,16 @@ pub trait INode: Any + Sync + Send {
         Err(FsError::NotSupported)
     }
 
+    /// Get the lock list of the INode, create an empty one if not exist
+    fn lock_list(&self) -> Result<Arc<dyn INodeLockList>> {
+        Err(FsError::NotSupported)
+    }
+
+    /// Test if INode has lock list
+    fn test_lock_list(&self) -> Option<Arc<dyn INodeLockList>> {
+        None
+    }
```

Comment:
I am not a big fan of the proposed changes to the `Inode` trait. Conceptually, lock lists are notions of OSes, not FSes. You also acknowledge this fact. That is why you have made an admirable effort to introduce the `INodeLockList` trait to decouple the implementations of lock lists (which is part of the OSes) from the interface (which is part of the VFS). But the result is not perfect in my opinion. The VFS layer stills needs to know the concept of `INodeLockList`, which is a trivial trait that virtually says nothing about lock lists, except that they are `Any`. Can we do better?

I think I have an alternative solution, which can completely remove the notion of lock lists from VFS. The solution is based on `Extensions`, which is a generalized and also extensible version of the current solution.

```rust
pub trait Inode {
    /// Get the extension of this inode.
    fn ext(&self) -> &Extension;
}

/// An extension is a set of objects that is attached to
/// an inode.
///
/// Each objects of an extension is of different types.
/// In other words, types are used as the keys to get and
/// set the objects in an extension.
//
// For implementation details, read [this blog post](https://www.jakobmeier.ch/blogging/Untapped-Rust.html)
// and see the [AnyMap](https://crates.io/crates/anymap) crate.
pub struct Extension(/* ... */);

impl Extension {
    pub fn new() -> Self {
        todo!("implement new")
    }

    /// Get an object of `Arc<T>`.
    pub fn get<T: Any + Send + Sync>(&self) -> Option<Arc<T>> {
        todo!("implement get")
    }

    /// Put an object of `Arc<T>`. If there exists one object of the type,
    /// then the old one is returned.
    pub fn put<T: Any + Send + Sync>(&self, obj: Arc<T>) -> Option<Arc<T>> {
        todo!("implement set")
    }

    /// Delete an object of `Arc<T>`. If there exists one object of the type,
    /// then the old one is returned.
    pub fn del<T: Any + Send + Sync>(&self) -> Option<Arc<T>> {
        todo!("implement del")
    }
}
```